kingfisher_scrapy/spiders/nigeria_cross_river_base.py

```diff
@@ -1,20 +1,33 @@
 from abc import abstractmethod
+from datetime import datetime
 
-from kingfisher_scrapy.base_spider import PeriodicSpider
-from kingfisher_scrapy.util import components, join, parameters
+import scrapy
 
+from kingfisher_scrapy.base_spider import SimpleSpider
+from kingfisher_scrapy.util import components, handle_http_error, join, parameters
 
-class NigeriaCrossRiverBase(PeriodicSpider):
+
+class NigeriaCrossRiverBase(SimpleSpider):
     # SimpleSpider
     base_url = 'http://ocdsapi.dppib-crsgov.org/api/ocdsAPI/'
 
     # BaseSpider
     date_format = 'year-month'
-    default_from_date = '2020-02'
+    default_from_date = '2021-02'
 
-    # PeriodicSpider
     formatter = staticmethod(join(components(-1), parameters('year', 'month')))
 
+    def start_requests(self):
+        url = 'http://ocdsapi.dppib-crsgov.org/api/ocdsAPI/getAvailableReleasesSummary'
+        yield scrapy.Request(url, meta={'file_name': 'list.json'})
+
+    @handle_http_error
+    def parse(self, response):
+        for item in response.json():
+            date = datetime(item['year'], item['month'], 1)
+            for number, url in enumerate(self.build_urls(date)):
+                yield self.build_request(url, formatter=self.formatter, callback=super().parse, priority=number * -1)
```

Comment:
If we change `build_url` to just return an URL, we can avoid this `for`, and also, for the `from_date` and `until_date` parameters to actually work, we need to check for them and filter the requests, something like:

```suggestion
            if self.from_date and self.until_date:
                if not (self.from_date <= date <= self.until_date):
                    continue
            yield self.build_request(self.build_url(date), formatter=formatter)
```